### PR TITLE
feat(indexer): reference caching

### PR DESCRIPTION
**Pull request by normanzb**
@michalkvasnicak can we break down https://linear.app/modprotocol/issue/ECP-1527/improve-comment-references-resolving-especially-urls into 2 parts:

In this PR we will only do the job related to cache references in a separate table.  While indexing if the comment's reference is not found in cache, the indexer fetches from the network and store it into the cache. 
Indexer itself does not handle expiry/failed/partial case, we will have async job separately to handle it later in separate PR.
While indexing, if reference fetching failed or cache is expired, let it be, it won't be the end of the world anyway, at most causing the client to not render properly temporarily and client should expect this to happen (e.g. due to network issue, url will be broken) and I assume this is going to be rare case anyway.

So in theory, when a comment comes in, the data will be fetched only once by the indexer.

- [x] add indexing b tree to status
- [x] add indexing for comment id

## Summary by CodeRabbit

- New Features
  - Cache-backed, revisioned storage and a cache-first API for comment reference resolution; centralized resolution service used by comment flows.

- Bug Fixes
  - Edge-case status logic adjusted: comments with no resolvable references are treated as successful rather than failed.

- Tests
  - Adds unit tests for cache and resolution services and updates indexing tests to verify cache behavior.

- Chores
  - Adds database migration and a frozen metadata snapshot.

- Documentation
  - Clarifies input-type behavior in SDK schema comments.

### CODE REVIEW

File: apps/indexer/src/services/types.ts [hunk 1]
```diff
@@ -330,3 +330,25 @@ export interface IAdminTelegramBotService {
   initialize: () => Promise<void>;
   handleWebhookRequest: Handler;
 }
+
+// Comment Reference Service/Resolvers Types
+/**
+ * Interface for comment reference resolver
+ */
+
```
Comment (michalkvasnicak) [inline apps/indexer/src/services/types.ts:338]: nit, extraneous line

File: apps/indexer/src/services/comment-references-cache-service.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          revision: schema.commentReferenceResolutionResults.revision,
+        })
+        .from(schema.commentReferenceResolutionResults)
+        .where(
+          eq(schema.commentReferenceResolutionResults.commentId, commentId),
+        )
+        .orderBy(desc(schema.commentReferenceResolutionResults.revision))
+        .limit(1);
+
+      const nextRevision = (latestResult[0]?.revision ?? -1) + 1;
```
Comment (michalkvasnicak) [inline apps/indexer/src/services/comment-references-cache-service.ts:64]: to be really sure here we should probably select FOR update

File: apps/indexer/src/services/types.ts (hunk 1 above)
Comment (normanzb) [reply apps/indexer/src/services/types.ts]: fixed

File: apps/indexer/src/services/comment-references-resolution-service.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      );
+
+      await this.commentReferencesCacheService.updateReferenceResolutionResult({
+        commentId,
+        commentRevision,
+        references: latestReferenceResult.references,
+        referencesResolutionStatus: latestReferenceResult.status,
+      });
+
+      return latestReferenceResult;
```
[the remark this answers is not in the document]
Comment (normanzb) [reply apps/indexer/src/services/comment-references-resolution-service.ts:97]: no need

File: apps/indexer/schema.offchain.ts [hunk 4]
```diff
@@ -56,6 +57,39 @@ export const commentClassificationResults = offchainSchema.table(
...
+    }),
+    check(
+      "comment_reference_resolution_status_enum",
+      sql`${table.referencesResolutionStatus} IN ('success', 'pending', 'partial', 'failed')`,
+    ),
+    index("comment_reference_resolution_results_by_status_idx").on(
+      table.referencesResolutionStatus,
+    ),
+  ],
+);
```
[the remark this answers is not in the document]
Comment (normanzb) [reply apps/indexer/schema.offchain.ts:88]: assert?
